src/main/java/com/thebois/models/beings/actions/DoNothingAction.java

```diff
@@ -0,0 +1,42 @@
+package com.thebois.models.beings.actions;
+
+import com.thebois.models.beings.IActionPerformer;
+
+/**
+ * Makes the performer do nothing at all.
+ * <p>
+ * This action never completes by itself. It is an action given to a being when it asks for an
+ * action but there is nothing more to do.
+ * </p>
+ */
+public class DoNothingAction implements IAction {
+
+    private boolean isDone = false;
+
+    @Override
+    public void perform(final IActionPerformer performer) {
+        isDone = true;
+    }
+
+    @Override
+    public boolean isCompleted(final IActionPerformer performer) {
+        return isDone;
+    }
```

Comment:
Just curious, is this if they are idle?

Reply:
Nope, when they're idle, they perform different `MoveAction`s with random destinations. This action is used when a `Role` wants to perform an actual `Action` but can't, for some reason (e.g. the tree it is trying to get to is blocked on all four sides). 

For how idle is handled, see `IdleRole`, and for how `DoNothingAction` is used, see `AbstractHarvestRole`.

Reply:
👍